week-4/E-arrays-of-objects/exercise-3.js

```diff
@@ -54,32 +54,57 @@ DO NOT EDIT ANYTHING ABOVE THIS LINE
 WRITE YOUR CODE BELOW
 */
 
-
 var restaurantFinderApplication = {
-    applicationName: "Restaurant Finder",
-    applicationVersion: "1.0",
-    restaurants: restaurants,
-    findAvailableRestaurants: function(numberOfPeople) {
-        // Complete here
-    },
-    findRestaurantServingDish: function(dishName) {
-        // Complete here
-    },
-    countNumberOfRestaurantsInArea: function(area) {
-        // Complete here
-    }
+  applicationName: "Restaurant Finder",
+  applicationVersion: "1.0",
+  restaurants: restaurants,
+  findAvailableRestaurants: function(numberOfPeople) {
+    // Complete here
+    return restaurants
+      .filter(restaurant => {
+        return (
+          restaurant.totalSeats - restaurant.numberOfCustomers >= numberOfPeople
+        );
+      })
+      .map(restaurant => restaurant.name);
+  },
+  findRestaurantServingDish: function(dishName) {
+    // Complete here
+    return restaurants
+      .filter(restaurant => restaurant.menu.includes(dishName))
+      .map(restaurant => restaurant.name);
+  },
+  countNumberOfRestaurantsInArea: function(area) {
+    // Complete here
+    return restaurants.filter(restaurant => restaurant.address.area === area)
+      .length;
+  }
 };
 
-
 /*
 DO NOT EDIT ANYTHING BELOW THIS LINE
 */
 
-var restaurantsAvailableFor5People = restaurantFinderApplication.findAvailableRestaurants(5);
-console.log("Find available restaurants for 5 people: Expected result: Ubiquitous Chip,Monkeyz, actual result: " + restaurantsAvailableFor5People);
-
-var restaurantsServingSalad = restaurantFinderApplication.findRestaurantServingDish("salad");
-console.log("Find restaurants serving salad: Expected result: Paesano,Ubiquitous Chip, actual result: " + restaurantsServingSalad);
-
-var numberOfRestaurantsInCityCentre = restaurantFinderApplication.countNumberOfRestaurantsInArea("center");
-console.log("Number of restaurants in city centre: Expected result: 2, actual result: " + numberOfRestaurantsInCityCentre);
+var restaurantsAvailableFor5People = restaurantFinderApplication.findAvailableRestaurants(
+  5
+);
+console.log(
+  "Find available restaurants for 5 people: Expected result: Ubiquitous Chip,Monkeyz, actual result: " +
+    restaurantsAvailableFor5People
+);
+
+var restaurantsServingSalad = restaurantFinderApplication.findRestaurantServingDish(
+  "salad"
+);
+console.log(
+  "Find restaurants serving salad: Expected result: Paesano,Ubiquitous Chip, actual result: " +
+    restaurantsServingSalad
+);
+
+var numberOfRestaurantsInCityCentre = restaurantFinderApplication.countNumberOfRestaurantsInArea(
+  "center"
+);
+console.log(
+  "Number of restaurants in city centre: Expected result: 2, actual result: " +
+    numberOfRestaurantsInCityCentre
+);
```

Comment:
Nice work! Good formatting and use of array methods here. I like how even though the question asks for `findRestaurantServingDish` to find dishes including `salad`, you've used the `dishName` parameter, so that it could work with other dishes as well 🙂